R/html_dependency.R

```diff
@@ -505,6 +518,68 @@ renderDependencies <- function(dependencies,
   HTML(paste(html, collapse = "\n"))
 }
 
+
+
+renderScript <- function(script, srcpath, encodeFunc, hrefFilter) {
```

Comment:
To make this change safer and easier to reason about, consider adding an early exit here, something like:

```r
if (is.character(script)) {
  return(renderScriptSrc(script, srcpath, encodeFunc, hrefFilter))
}
```

With `renderScriptSrc()` defined as:

```r
renderScriptSrc <- function(script, path, encodeFunc, hrefFilter) {
  paste0(
    '<script src="',
    htmlEscape(hrefFilter(file.path(path, encodeFunc(script)))),
    "'></script>"
  )
}
```

You could then also reuse this function when you go to render the more complex list case

Reply:
I don't really see how that could be used for the more complex list case.

The llist case aims to support instances such as:

```r
renderScript(list("s1.js", list(src = "s2.js"), list(src = "s3.js", integrity = "hash", crossorigin = "anonymous")))
```

```html
<script src="s1.js"></script>
<script src="s2.js"></script>
<script src="s3.js" integrity="hash" crossorigin="anonymous"></script>
```

or:

```r
renderScript(list(src = "s3.js", integrity = "hash", crossorigin = "anonymous"))
```

```html
<script src="s3.js" integrity="hash" crossorigin="anonymous"></script>
```

These accept mixed use of character vectors, nested lists, & pure lists.

Unless I'm mistaken, the change you're proposing would mean I'd have to duplicate the actual templating pattern for the case of unnamed character vectors, and named lists.